src/snkit/network.py

```diff
@@ -254,19 +260,49 @@ def snap_node(geom):
     return Network(nodes=nodes, edges=network.edges)
 
 
-def split_edges_at_nodes(network, tolerance=1e-9):
-    """Split network edges where they intersect node geometries"""
+def _split_edges_at_nodes(edges: GeoDataFrame, nodes: GeoDataFrame, tolerance: float) -> list:
+    """Split edges at nodes for a network chunk"""
     split_edges = []
-    for edge in tqdm(
-        network.edges.itertuples(index=False), desc="split", total=len(network.edges)
-    ):
-        hits = nodes_intersecting(edge.geometry, network.nodes, tolerance)
+
+    for edge in edges.itertuples():
+        hits = nodes_intersecting(edge.geometry, nodes, tolerance)
         split_points = MultiPoint([hit.geometry for hit in hits.itertuples()])
 
         # potentially split to multiple edges
         edges = split_edge_at_points(edge, split_points, tolerance)
         split_edges.append(edges)
 
+    return split_edges
+
+
+def split_edges_at_nodes(network, tolerance=1e-9):
+    """Split network edges where they intersect node geometries"""
+    split_edges = []
+
+    n = len(network.edges)
+    if PARALLEL and (n > 10_000):
+        chunk_size = int(n / os.cpu_count())
+        args = [
+            (network.edges.iloc[i: i + chunk_size, :], network.nodes, tolerance)
+            for i in range(0, n, chunk_size)
+        ]
+        with multiprocessing.Pool() as pool:
+            results = pool.starmap(_split_edges_at_nodes, args)
+
+        # flatten return list
+        split_edges: list = [df for chunk in results for df in chunk]
+
+    else:
+        for edge in tqdm(
+            network.edges.itertuples(index=False), desc="split", total=len(network.edges)
+        ):
+            hits = nodes_intersecting(edge.geometry, network.nodes, tolerance)
+            split_points = MultiPoint([hit.geometry for hit in hits.itertuples()])
+
+            # potentially split to multiple edges
+            edges = split_edge_at_points(edge, split_points, tolerance)
+            split_edges.append(edges)
```

Comment:
The `else` case here could be a single call to `_split_edges_at_nodes` to avoid repetition.